x-pack/legacy/plugins/maps/public/layers/sources/client_file_source/geojson_file_source.js

```diff
@@ -53,6 +53,7 @@ export class GeojsonFileSource extends AbstractVectorSource {
     return (indexResponses = {}) => {
       const { indexDataResp, indexPatternResp } = indexResponses;
       if (!(indexDataResp && indexDataResp.success) ||
```

Comment:
can you split these three conditions up and assign to variable with human readable name. Especially the new one introduced here would benefit from

```
const indexCreationFailed = !(indexDataResp && indexDataResp.success);
const allDocsFailed = indexDataResp.failures.length === indexDataResp.docCount;
const indexPatternCreationFailed =  !(indexPatternResp && indexPatternResp.success);
```

Reply:
Good call. That was getting pretty cluttered